contracts/external/oracles/DaiPriceOracle.sol

```diff
@@ -280,20 +265,25 @@ contract DaiPriceOracle is
     }
 
     /**
-     * Get the USD price of DAI according to Uniswap given the USD price of ETH.
+     * Get the DAI-USDC price according to Uniswap.
      *
-     * @param  ethUsd  Monetary.Price struct with a fixed-point value with 18 decimals of precision.
-     * @return         Monetary.Price struct with a fixed-point value with 18 decimals of precision.
+     * @return  Monetary.Price struct with a fixed-point value with 18 decimals of precision.
      */
-    function getUniswapPrice(
-        Monetary.Price memory ethUsd
-    )
+    function getUniswapPrice()
         public
         view
         returns (Monetary.Price memory)
     {
-        (uint256 daiAmt, uint256 ethAmt, ) = UNISWAP.getReserves();
-        uint256 uniswapPrice = Math.getPartial(ethUsd.value, ethAmt, daiAmt);
+        (uint256 daiAmt, uint256 poolOneEthAmt, ) = UNISWAP_DAI_ETH.getReserves();
+        (uint256 usdcAmt, uint256 poolTwoEthAmt, ) = UNISWAP_USDC_ETH.getReserves();
+
+        // Get the price of DAI in USDC. Multiply by 10^30 to account for the difference in deicmals
+        // between DAI and USDC, and get a result with 18 decimals of precision.
+        uint256 uniswapPrice = UNISWAP_DECIMALS_BASE
+            .mul(usdcAmt)
+            .mul(poolOneEthAmt)
```

Comment:
I believe there's potential for overflow here for some tokens.
I think it should be fine for USDC since it has only 6 decimals.

10^30 * 10^6 * 10^18 for 1 USDC and 1 ETH

Maybe something to keep in mind or at least comment

edit: I see you have a test for this which is great